add passthrough filter chain support for v1beta1 authorization policy

yangminzhu opened the pull request:
For #12394

1. Added e2e tests with v1beta1 authorization policy for the passthrough filter chain.
1. Tested manually in the cluster that the RBAC filter is inserted for the passthrough filter chain.


And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

istio-testing commented:
@yangminzhu: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2074) | `/test pilot-multicluster-e2e_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2074) | `/test pilot-multicluster-e2e_istio`
lint_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2825) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2074) | `/test pilot-multicluster-e2e_istio`
lint_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2825) | `/test lint_istio`
integ-galley-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-galley-k8s-tests_istio/2834) | `/test integ-galley-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2074) | `/test pilot-multicluster-e2e_istio`
lint_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2825) | `/test lint_istio`
integ-galley-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-galley-k8s-tests_istio/2834) | `/test integ-galley-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2155) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2074) | `/test pilot-multicluster-e2e_istio`
lint_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2825) | `/test lint_istio`
integ-galley-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-galley-k8s-tests_istio/2834) | `/test integ-galley-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2155) | `/test integ-new-install-k8s-tests_istio`
integ-mixer-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2229) | `/test integ-mixer-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

lambdai commented on pilot/pkg/networking/core/v1alpha3/listener_builder.go:
```diff
@@ -439,21 +439,33 @@ func newInboundPassthroughFilterChains(env *model.Environment, node *model.Proxy
+		if b := builder.NewBuilder(env.Mesh.TrustDomain, env.Mesh.TrustDomainAliases, nil,
+			node.WorkloadLabels, node.ConfigNamespace, push.AuthzPolicies, isXDSMarshalingToAnyEnabled); b != nil {
+			if authzFilter := b.BuildTCPFilter(); authzFilter != nil {
+				filterChain.Filters = []*listener.Filter{
```
nit: append instead of literal slice

yangminzhu replied:
The authzFilter needs to be added in the beginning of the slice, do you mean something like the following?
```
filterChain.Filters = append([]*listener.Filter{authzFilter}, filterChain.Filters...)
```

yangminzhu commented:
> Right now don't we only call rbac/mixer code through the plug-in interface? directly calling rbac here seems like the wrong place since it breaks the plug-in abstraction

Switched to use plugin interface.

@lambdai Also added an e2e test.

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
pilot-multicluster-e2e_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2074) | `/test pilot-multicluster-e2e_istio`
integ-galley-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-galley-k8s-tests_istio/2834) | `/test integ-galley-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2155) | `/test integ-new-install-k8s-tests_istio`
integ-mixer-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2229) | `/test integ-mixer-k8s-tests_istio`
lint_istio | 4fa92c7947b5c3108493430b7e1a98d43289e7f8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2884) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
integ-galley-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-galley-k8s-tests_istio/2834) | `/test integ-galley-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2155) | `/test integ-new-install-k8s-tests_istio`
integ-mixer-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2229) | `/test integ-mixer-k8s-tests_istio`
lint_istio | 4fa92c7947b5c3108493430b7e1a98d43289e7f8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2884) | `/test lint_istio`
pilot-multicluster-e2e_istio | 7ad9ead619874ca88cb39d8cf377fb6b8d08a0d4 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2139) | `/test pilot-multicluster-e2e_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/unit-tests_istio/2944) | `/test unit-tests_istio`
integ-galley-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-galley-k8s-tests_istio/2834) | `/test integ-galley-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2155) | `/test integ-new-install-k8s-tests_istio`
integ-mixer-k8s-tests_istio | 8881a441abd8d638138ace333a9da90f28960d11 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2229) | `/test integ-mixer-k8s-tests_istio`
pilot-multicluster-e2e_istio | 7ad9ead619874ca88cb39d8cf377fb6b8d08a0d4 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2139) | `/test pilot-multicluster-e2e_istio`
lint_istio | 7ad9ead619874ca88cb39d8cf377fb6b8d08a0d4 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2888) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented on pilot/pkg/networking/core/v1alpha3/listener_builder.go:
```diff
@@ -406,8 +407,32 @@ func newBlackholeFilter(enableAny bool) *listener.Filter {
+	}
+	if err := authz.NewPlugin().OnInboundListener(in, mutable); err == nil {
+		if protocol == plugin.ListenerProtocolTCP && len(mutable.FilterChains[0].TCP) == 1 {
+			return mutable.FilterChains[0].TCP[0]
```
Do we need bounds checks here? This seems fragile?

yangminzhu replied:
`mutable.FilterChains` is guaranteed to have at least 1 element from line 420

howardjohn commented on pilot/pkg/networking/core/v1alpha3/listener_builder.go:
```diff
@@ -406,8 +407,32 @@ func newBlackholeFilter(enableAny bool) *listener.Filter {
+			},
+		},
+	}
+	if err := authz.NewPlugin().OnInboundListener(in, mutable); err == nil {
```
this still doesn't feel right.. shouldn't we just add `OnInboundPasssthrough` or something similar?

Even if we are calling the plugin here, this still breaks the plugin model. Core pilot code should not know about plugins, it should just have a list of plugins the user turns on and call callbacks for them. Authz may not even be enabled but we are using the authz plugin regardless

yangminzhu replied:
I think it could be a long term solution to add something like `OnInboundPasssthrough`. But I'm afraid this needs more design discussions, are we okay that this new plugin interface is only used for the passthrough filter chain? That sounds too limited use case and seems we're adding it only for solving the problem here. 

Instead, we may want to add it in a more general way that could better work for other use cases as well, this may need more design.   cc @lambdai 

Here I'm trying to do a short-term fix to only insert the authz filter in the new code path because authz is the only filter that needs to be inserted here, and I also want to avoid any side-effect of adding other filters from other plugins.

The authz plugin is enabled by default, I'm not sure in what case the customer would want to disable the plugin, they could just remove all their authz policies for the same effect. If we're really worried it may be disabled by some users, i can do some refactor to pass around the enabled plugin lists but this will require some more code changes in the other places in pilot.

howardjohn replied:
I don't think there is any point in having plugins at all if we are just going to call them directly from core pilot code. The whole intent of them is to keep separate concerns isolated. The other plugins can just do nothing `OnInboundPasssthrough`, so I don't think that is a concern.

yangminzhu replied:
@howardjohn Cool, updated to use `OnInboundPasssthrough`, PTAL, thanks!

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pilot-multicluster-e2e_istio | 7ad9ead619874ca88cb39d8cf377fb6b8d08a0d4 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2139) | `/test pilot-multicluster-e2e_istio`
integ-new-install-k8s-tests_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2221) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2221) | `/test integ-new-install-k8s-tests_istio`
pilot-multicluster-e2e_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2142) | `/test pilot-multicluster-e2e_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2221) | `/test integ-new-install-k8s-tests_istio`
pilot-multicluster-e2e_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2142) | `/test pilot-multicluster-e2e_istio`
lint_istio | e707fd7a64429afb94ae45ff3b16da0b11203112 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2911) | `/test lint_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2221) | `/test integ-new-install-k8s-tests_istio`
lint_istio | e707fd7a64429afb94ae45ff3b16da0b11203112 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/lint_istio/2911) | `/test lint_istio`
pilot-multicluster-e2e_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2162) | `/test pilot-multicluster-e2e_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 9264ba1438ae6e5a42c43ef16c1cf4fe4e7c5982 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2221) | `/test integ-new-install-k8s-tests_istio`
pilot-multicluster-e2e_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2162) | `/test pilot-multicluster-e2e_istio`
integ-mixer-k8s-tests_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2306) | `/test integ-mixer-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pilot-multicluster-e2e_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2162) | `/test pilot-multicluster-e2e_istio`
integ-mixer-k8s-tests_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2306) | `/test integ-mixer-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2242) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

rshriram commented on pilot/pkg/networking/core/v1alpha3/listener_builder.go:
```diff
@@ -439,20 +439,38 @@ func newInboundPassthroughFilterChains(env *model.Environment, node *model.Proxy
+			},
+		}
+		for _, p := range configgen.Plugins {
+			if err := p.OnInboundPasssthrough(in, mutable); err != nil {
```
Typo. sss

yangminzhu replied:
Fixed, thanks!

liminw commented on pilot/pkg/networking/plugin/authz/authorization.go:
```diff
@@ -168,3 +163,14 @@ func (Plugin) OnInboundRouteConfiguration(in *plugin.InputParams, route *xdsapi.
+
+// OnInboundPasssthrough is called whenever a new passthrough filter chain is added to the LDS output.
+func (Plugin) OnInboundPasssthrough(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
+	if in.Node.Type != model.SidecarProxy {
```
Does pass through filter exist on Ingress/Egress?

yangminzhu replied:
No, only on Sidecar.

istio-testing commented:
@yangminzhu: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pilot-multicluster-e2e_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/pilot-multicluster-e2e_istio/2162) | `/test pilot-multicluster-e2e_istio`
integ-mixer-k8s-tests_istio | 2164ca1460415bda5538f60e562d0599ec939848 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-mixer-k8s-tests_istio/2306) | `/test integ-mixer-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 66b7b82b61d2a50a03ef4f4a731b23da2062dddf | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2282) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

yangminzhu commented:
@howardjohn @rshriram @lambdai Could you approve if there is no change needed? Thanks.

lambdai commented:
Generally LGTM

@yxue could you share some light on how tcp service is tested on default chain?

I didn't catch up all the progress. My impression was left at the age that echo client will always be sniffed as http since the traffic pattern at echo client is always Http-like.

howardjohn commented:
integ-new-install-k8s-tests_istio is failing on the new test

yangminzhu commented:
/test integ-new-install-k8s-tests_istio

istio-testing commented:
@yangminzhu: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 66b7b82b61d2a50a03ef4f4a731b23da2062dddf | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2367) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

yangminzhu commented:
The test failed with timeout:
```
=== RUN   TestMirroring
2019-10-25T22:48:12.163262Z	info	pickfirstBalancer: HandleSubConnStateChange: 0xc00062bdd0, CONNECTING
2019-10-25T22:48:12.170807Z	info	pickfirstBalancer: HandleSubConnStateChange: 0xc00062bdd0, READY
2019-10-25T22:48:22.665491Z	info	transport: loopyWriter.run returning. connection error: desc = "transport is closing"
panic: test timed out after 30m0s
```

Let me rerun it one again /test integ-new-install-k8s-tests_istio

yangminzhu commented:
/test integ-new-install-k8s-tests_istio

yangminzhu commented:
@howardjohn The `integ-new-install-k8s-tests_istio` is now failing for time out on many different PRs: https://k8s-testgrid.appspot.com/istio_istio#integ-new-install-k8s-tests_istio&width=20, I suspect there is something else causing the timeout, can we merge this PR first?

howardjohn commented:
@yangminzhu understood, that will need to be fixed. I am working on making the test required.

Your PR is introducing an additional failure though. So we can merge if that passes, but not add a new error

yangminzhu commented:
@howardjohn How do you know this PR is introducing new error for the test?

Also filed https://github.com/istio/istio/issues/18351 for the test failing.

howardjohn commented:
Your PR adds TestV1beta1_PassThroughFilterChain which failed in https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2282 and https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2242, or is that just an old commit I was looking at maybe?

istio-testing commented:
@yangminzhu: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 73e6d9aee98e2d76b9bf815f59f534068f41f7d1 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2377) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

yangminzhu commented:
/test integ-new-install-k8s-tests_istio

yangminzhu commented:
> Your PR adds TestV1beta1_PassThroughFilterChain which failed in https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2282 and https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2242, or is that just an old commit I was looking at maybe?

I suspect these are from some old commits, I tried to run `make test.integration.new.installer` with a small customization to the configmap to remove the `applicationPorts` to workaround the other issue, The `TestV1beta1_PassThroughFilterChain` is passing with it.

istio-testing commented:
@yangminzhu: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-tests_istio | 73e6d9aee98e2d76b9bf815f59f534068f41f7d1 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/18207/integ-new-install-k8s-tests_istio/2381) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented:
/retest

yangminzhu commented:
@howardjohn It seems istio-testing doesn't create the cherrypick to release-1.4 automatically for this PR, will create one manually.

howardjohn commented:
/cherrypick release-1.4

istio-testing commented:
@howardjohn: new pull request created: #18409

<details>

In response to [this](https://github.com/istio/istio/pull/18207#issuecomment-547164889):

>/cherrypick release-1.4


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>
